.github/workflows/docker.yml

```diff
@@ -23,12 +29,91 @@ jobs:
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}    
+   
     - name: Set up environment
       run: |
         export GOPATH=$(go env GOPATH)
         echo "GOPATH=$GOPATH" >> $GITHUB_ENV
         echo "$GOPATH/bin" >> $GITHUB_PATH
 
-    - name: Build images
-      run: |
-        make -C dist/images all
+    - name: build ovnkube-binaries copy to context 
+      run: | 
+        pushd go-controller 
+         make 
+        popd 
+
+        pushd dist/images
+          cp -r ../../go-controller/_output/go/bin/* .
+        popd 
+    
+    - name: generate git-info to write to image
+      run: | 
+        BRANCH=$(git rev-parse --short "$GITHUB_SHA")
+        COMMIT=${GITHUB_REF#refs/heads/}
+        pushd dist/images
+          echo "ref: ${BRANCH}  commit: ${COMMIT}" > git_info
+        popd 
+    
+    - name: Extract metadata (tags, labels) for fedora ovn-k image
+      id: meta-fedora
+      uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+      with:
+        images: ${{ env.REGISTRY }}/${{ env.OWNER }}/${{ env.REPOSITORY }}/${{ env.FEDORA_IMAGE_NAME }}
+
+    - name: Build and push Fedora based Docker image
+      uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+      with:
+        context: ./dist/images
+        file: ./dist/images/Dockerfile.fedora
+        push: true
+        tags: ${{ steps.meta-fedora.outputs.tags }}
+        labels: ${{ steps.meta-fedora.outputs.labels }}
+
+    - name: Extract metadata (tags, labels) for ubuntu ovn-k image
+      id: meta-ubuntu
+      uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+      with:
+        images: ${{ env.REGISTRY }}/${{ env.OWNER }}/${{ env.REPOSITORY }}/${{ env.UBUNTU_IMAGE_NAME }}
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@master
+      with:
+        platforms: all
+
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@master
+      
+    - name: Build and push Ubuntu based Docker image
+      uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+      with:
+        builder: ${{ steps.buildx.outputs.name }}
+        context: ./dist/images
+        file: ./dist/images/Dockerfile.ubuntu
+        push: true
+        platforms: linux/amd64,linux/arm64
+        tags: ${{ steps.meta-ubuntu.outputs.tags }}
+        labels: ${{ steps.meta-ubuntu.outputs.labels }}
+
+    - name: Extract metadata (tags, labels) for centos ovn-k image
+      id: meta-centos
+      uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+      with:
+        images: ${{ env.REGISTRY }}/${{ env.OWNER }}/${{ env.REPOSITORY }}/${{ env.CENTOS_IMAGE_NAME }}
+    
+    - name: Build and push Centos based Docker image
+      uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+      with:
+        context: ./dist/images
+        file: ./dist/images/Dockerfile
+        push: true
```

Comment:
I know in the docs you say centos doesnt have arm build, but why not?

Reply:
This is just what we did before I will look into adding arm builds for fedora